Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/submission/RatProfileCensor.kt

```diff
@@ -0,0 +1,64 @@
+package de.rki.coronawarnapp.bugreporting.censors.submission
+
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.toNewLogLineIfDifferent
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidAddress
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidCity
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidName
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidPhoneNumber
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidZipCode
+import de.rki.coronawarnapp.bugreporting.debuglog.LogLine
+import de.rki.coronawarnapp.coronatest.antigen.profile.RATProfileSettings
+import kotlinx.coroutines.flow.first
+import org.joda.time.format.DateTimeFormat
+import javax.inject.Inject
+
+class RatProfileCensor @Inject constructor(
+    private val ratProfileSettings: RATProfileSettings
+) : BugCensor {
+
+    override suspend fun checkLog(entry: LogLine): LogLine? {
+        val ratProfile = ratProfileSettings.profile.flow.first() ?: return null
+
+        var newMessage = entry.message
+
+        with(ratProfile) {
+            withValidName(firstName) { firstName ->
+                newMessage = newMessage.replace(firstName, "RAT-Profile/FirstName")
+            }
+
+            withValidName(lastName) { lastName ->
+                newMessage = newMessage.replace(lastName, "RAT-Profile/LastName")
+            }
+
+            val formatter = DateTimeFormat.forPattern("yyyy-MM-dd")
```

Comment:
Please move this to class level or a companion object, `checkLog` will be called for every line, creating a new  `DateTimeFormat` instance for every logline is costly :money_with_wings: